content/docs/iac/using-pulumi/extending-pulumi/use-terraform-module.md

```diff
@@ -0,0 +1,283 @@
+
+***Example:** index.ts - Using the Terraform VPC and RDS module in a Pulumi program*
+
+```typescript
+import * as vpcmod from '@pulumi/vpcmod';
+import * as pulumi from '@pulumi/pulumi';
+import * as rdsmod from '@pulumi/rdsmod';
+import * as aws from '@pulumi/aws';
+import * as std from '@pulumi/std';
+
+// Get available availability zones
+const azs = aws.getAvailabilityZonesOutput({
+  filters: [{
+        name: "opt-in-status",
+        values: ["opt-in-not-required"],
+    }]
+}).names.apply(names => names.slice(0, 3));
+
+const cidr = "10.0.0.0/16";
+
+const cfg = new pulumi.Config();
+const prefix = cfg.get("prefix") ?? pulumi.getStack();
+
+// Create a VPC using the terraform-aws-modules/vpc module
+const vpc = new vpcmod.Module("test-vpc", {
+  azs: azs,
+  name: `test-vpc-${prefix}`,
+  cidr,
+  public_subnets: azs.apply(azs => azs.map((_, i) => {
+    return getCidrSubnet(cidr, i+1);
+  })),
+  private_subnets: azs.apply(azs => azs.map((_, i) => {
+    return getCidrSubnet(cidr, i+1+4);
+  })),
+  database_subnets: azs.apply(azs => azs.map((_, i) => {
+    return getCidrSubnet(cidr, i+1 + 8);
+  })),
+  create_database_subnet_group: true,
+});
+
+// Create a security group for the RDS instance
+const rdsSecurityGroup = new aws.ec2.SecurityGroup('test-rds-sg', {
+  vpcId: vpc.vpc_id.apply(id => id!),
+});
+
+new aws.vpc.SecurityGroupIngressRule('test-rds-sg-ingress', {
+  ipProtocol: 'tcp',
+  securityGroupId: rdsSecurityGroup.id,
+  cidrIpv4: vpc.vpc_cidr_block.apply(cidr => cidr!),
+  fromPort: 3306,
+  toPort: 3306,
+});
+
+// Create an RDS instance using the terraform-aws-modules/rds module
+new rdsmod.Module("test-rds", {
+  engine: "mysql",
+  identifier: `test-rds-${prefix}`,
+  manage_master_user_password: true,
+  publicly_accessible: false,
+  allocated_storage: 20,
+  max_allocated_storage: 100,
+  instance_class: "db.t4g.large",
+  engine_version: "8.0",
+  family: "mysql8.0",
+  db_name: "completeMysql",
+  username: "complete_mysql",
+  port: '3306',
+  multi_az: true,
+  db_subnet_group_name: vpc.database_subnet_group_name.apply(name => name!),
+  vpc_security_group_ids: [rdsSecurityGroup.id],
+  skip_final_snapshot: true,
+  deletion_protection: false,
+  create_db_option_group: false,
+  create_db_parameter_group: false,
+});
+
+// Utility function to calculate subnet CIDRs
+function getCidrSubnet(cidr: string, netnum: number): pulumi.Output<string> {
+    return std.cidrsubnetOutput({
+    input: cidr,
+    newbits: 8,
+    netnum,
+  }).result;
+}
+```
+
+In the above code, the imported Terraform module works the same as any other Pulumi code. Outputs are returned, and resource state is stored in your Pulumi state storage, alongside all your other Pulumi-native resources. This also means that resource dependencies work as expected between Pulumi-native resources and resources created by Terraform modules.
+
+## Configuring Terraform Providers
```

Comment:
It can be confusing that Terraform providers are not the same as Pulumi bridged providers and need to be separately configured.

One cool feature possibly worth mentioning is that if you do have a Pulumi bridged provider configured you can reuse that configuration for the Terraform provider as well.  AS of today this works:


```
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as vpc from "@pulumi/vpc";

const awsProvider = new aws.Provider("aws-provider", {region: "us-west-2"});

const provider = new vpc.Provider("test-provider", {
    aws: awsProvider.terraformConfig().result,
});

// Use the provider with the module
const testBucket = new vpc.Module("test-bucket", {
    azs: [],
}, { provider: provider });

```